paint/PAINT_ColourPicker.cpp

```diff
@@ -77,34 +109,74 @@ void ColourPicker::setPrimarySecondaryFromActiveDraw() const
 	}
 }
 
+
+void ColourPicker::swapActiveColour() const
+{
+	primaryColourDisplay_->swapIsActive();
+	secondaryColourDisplay_->swapIsActive();
+
+	setDrawColourFromActive();
+	//update();
+
+	//rerender();
+}
+
+void ColourPicker::updateColourDisplaysFromDrawWindow() const
+{
+	const auto primaryRGBA = drawWindowPtr_->getPrimaryRGBA();
+	const auto secondaryRGBA = drawWindowPtr_->getSecondaryRGBA();
+	if (!swappedDisplays_) {
+		//primaryColourDisplay_->setColour(drawWindowPtr_->getPrimaryColour());
+		//secondaryColourDisplay_->setColour(drawWindowPtr_->getSecondaryColour());
+		primaryColourDisplay_->setColour(gfx::Colour(primaryRGBA[0], primaryRGBA[1], primaryRGBA[2], primaryRGBA[3]));
+		secondaryColourDisplay_->setColour(gfx::Colour(secondaryRGBA[0], secondaryRGBA[1], secondaryRGBA[2], secondaryRGBA[3]));
+	}
+	else {
+		//primaryColourDisplay_->setColour(drawWindowPtr_->getSecondaryColour());
+		//secondaryColourDisplay_->setColour(drawWindowPtr_->getPrimaryColour());
+		primaryColourDisplay_->setColour(gfx::Colour(secondaryRGBA[0], secondaryRGBA[1], secondaryRGBA[2], secondaryRGBA[3]));
+		secondaryColourDisplay_->setColour(gfx::Colour(primaryRGBA[0], primaryRGBA[1], primaryRGBA[2], primaryRGBA[3]));
+	}
+}
+
+void ColourPicker::updateColourDisplays() const
+{
+	setPrimarySecondaryFromActiveDraw();
+	primaryColourDisplay_->draw();
+	secondaryColourDisplay_->draw();
+	SDL_RenderPresent(renderer_);
+}
+
 void ColourPicker::updateColourValueBoxes() const
 {
-	printf("Updating RGBA\n");
 	redValueBox_->valueChangedExternally();
 	greenValueBox_->valueChangedExternally();
 	blueValueBox_->valueChangedExternally();
 	alphaValueBox_->valueChangedExternally();
 }
 
-void ColourPicker::swapActiveColour()
+void ColourPicker::updateColourSliders() const
 {
-	primaryColourDisplay_->swapIsActive();
-	secondaryColourDisplay_->swapIsActive();
+	redValueSlider_->valueChangedExternally();
+	greenValueSlider_->valueChangedExternally();
+	blueValueSlider_->valueChangedExternally();
+	alphaValueSlider_->valueChangedExternally();
+}
 
-	setDrawColourFromActive();
+void ColourPicker::update()
+{
+	updateColourDisplays();
 	updateColourValueBoxes();
+	updateColourSliders();
 }
 
-void ColourPicker::updateColourDisplaysFromDrawWindow()
```

Comment:
So is there a way you can decouple the colour picker from the draw window? The dependency seems a bit "top-heavy". Perhaps we need to add a new class called a ColourPalette which is shared by the draw window and this so that they can communicate?
**COULD**